src/scope.rs

```diff
@@ -1280,8 +1280,8 @@ pub(crate) mod data {
         .map(NonNull::as_ptr)
         .map(|p| unsafe { &mut *p })
         .unwrap();
-      match self_mut.status.get() {
-        ScopeStatus::Current { .. } => self_mut,
+      match self_mut.status {
+        ScopeStatus::CurrentZombie | ScopeStatus::CurrentNoZombie => self_mut,
         _ => unreachable!(),
       }
```

Comment:
While we're in here, can we convert this match/unreachable pattern to a `debug_assert`?